src/huggingface_hub/keras_mixin.py

```diff
@@ -342,48 +385,115 @@ def push_to_hub_keras(
     Returns:
         The url of the commit of your model in the given repository.
     """
+    if repo_id is not None:
+        token, _ = hf_api._validate_or_retrieve_token(token)
+        api = HfApi(endpoint=api_endpoint)
+
+        api.create_repo(
+            repo_id=repo_id,
+            repo_type="model",
+            token=token,
+            private=private,
+            exist_ok=True,
+        )
 
-    token, _ = hf_api._validate_or_retrieve_token(token)
-    api = HfApi(endpoint=api_endpoint)
+        # Push the files to the repo in a single commit
+        with tempfile.TemporaryDirectory() as tmp:
+            saved_path = Path(tmp) / repo_id
+            save_pretrained_keras(
+                model,
+                saved_path,
+                config=config,
+                include_optimizer=include_optimizer,
+                tags=tags,
+                plot_model=plot_model,
+                **model_save_kwargs,
+            )
 
-    api.create_repo(
-        repo_id=repo_id,
-        token=token,
-        private=private,
-        repo_type=None,
-        exist_ok=True,
-    )
+            if log_dir is not None:
+                # TODO: delete log folder on the hub when it exists
+                # previous implementation using git was doing it in the same git commit
+                # ... to be continued
+                tmp_log_dir = saved_path / "logs"
+                if os.path.exists(tmp_log_dir):
+                    rmtree(tmp_log_dir)
+                copytree(log_dir, tmp_log_dir)
```

Comment:
Do not forget to remove.

My strategy here would be to use `create_commit` instead of `upload_folder`. This would allow to have a Delete operation along the AddFile operations. Since we are in a temporary directory we can safely list of new files.
I am just not sure if we can delete an entire folder using `create_commit`.

Reply:
I think it could be done with `create_commit` indeed! I'd try a few approaches on a dummy repository to find an approach you're satisfied with. I think a single HTTP call to see what files there are and if there's an existing log dir isn't too expensive so I wouldn't be against having that implemented if you think that's the cleanest.

Reply:
I'll try that indeed. If it's not considered too expensive, I'll indeed do something like that with an extra call.

Reply:
Addressed in [bea0b77](https://github.com/LysandreJik/huggingface_hub/pull/8/commits/bea0b775121ff15206c383e293025e5a9d437e09). 

I am not a huge fan of my implementation since it duplicates some code from `upload_folder`. The problem I had is that I want to do exactly what `upload_folder` + add some Delete operations to delete log files in the same commit. I saw 3 solutions here but none were appealing to me:
1. This one. Refactor `upload_folder` and then duplicate 3 lines of code. "It works" but if we patch `upload_folder` in the future, we need to patch the keras mixin as well. 
2. Create 2 commits when using "push to hub": 1 to delete logs, 1 to push the model. I am not a big fan of this one: what is issues between 2 commits ? Can it work with create_pr/custom branch revision ?
3. Add an optional argument `additional_operations` to `upload_folder` to add additional operations to the "upload folder commit". I am not a big fan since I feel it's a quite custom need and `upload_folder` is meant to be a public method.

@LysandreJik If you have any thoughts on how this could be done better, I am interested. Solution 1 is implemented but I am fine with implementing solution 2 or 3 (or any other solution) if you are opinionated about this.

Reply:
I think 2 is unideal due to the two commits, and I think it would be quite hard to design an intuitive API for 3. So 1 is fine for me.

I think the changes done in bea0b77 are good but when reading the code it's a bit hard to understand what is the `_generate_url` and why it's used here. I don't understand what is the `content_type` and breaks a bit the flow of code when reading the method.

In that situation I'd either refactor that method to be super clear and understandable, or just keep the copy-paste which is also completely fine with me as it's (imo) more understandable this way and it's only there in a few spots.